### PR TITLE
MGMT-12439: Add basic job to test heterogeneous infrastructure

### PR DESCRIPTION
**Pull request by adriengentil**
Add first building blocks to test assisted with heterogeneous clusters.
It will first be used to test
https://github.com/openshift/assisted-test-infra/pull/1949.

### CODE REVIEW

File: ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  steps:
+    cluster_profile: packet-assisted
+    allow_best_effort_post_steps: true
+    allow_skip_on_success: true
+    pre:
+      - ref: assisted-equinix-setup
+      - chain: assisted-common-pre
+    test:
+      - ref: assisted-baremetal-conf
+      - ref: assisted-baremetal-test
```
Comment (osherdp) [inline ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:12]: (not necessarily in the scope of this PR) in the future we might want to move to ``assisted-common-verify``

File: ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    post:
+      - ref: assisted-common-gather
```
Comment (osherdp) [inline ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:14]: am I wrong or we're not tearing down the machines?

Comment (osherdp) [inline ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:14]: should we add ``baremetalds-packet-gather-metadata``?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml (hunk 1 above)
Comment (adriengentil) [reply ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:12]: Yes, we already have a ticket for that (assigned to me :see_no_evil: )

File: ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml (hunk 2 above)
Comment (adriengentil) [reply ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:14]: I'll need to create ticket about that, we now have to 2 machines to report, I don't know yet how to handle the case.

Comment (adriengentil) [reply ci-operator/step-registry/assisted/equinix/assisted-equinix-workflow.yaml:14]: :man_facepalming:

**Review by osherdp — APPROVED** [review]

**Pull request merged**